File: Weekly_Contests/Biweekly_99/Q1.cpp
class Solution {
public:
    int cal(vector<int>& nums){
        int res = 0;
        for (int i = 0; i < nums.size(); i++){
            res = res * 10 + nums[i];
        }
        return res;
    }

    int splitNum(int num) {
        vector<int> nums;
        while (num){
            nums.push_back(num % 10);
            num /= 10;
        }
        sort(nums.begin(), nums.end());
        vector<int> n1, n2;
        for (int i = 0; i < nums.size(); i++){
            if (i % 2 == 0){
                n1.push_back(nums[i]);
            }
            else{
                n2.push_back(nums[i]);
            }
        }
        return cal(n1) + cal(n2);
    }
};

File: Weekly_Contests/Biweekly_99/Q1.py
class Solution:
    def splitNum(self, num: int) -> int:
        digits = []
        while num:
            digits.append(num % 10)
            num //= 10
        digits.sort()
        n1, n2 = [], []
        for i in range(len(digits)):
            if i % 2 == 0:
                n1.append(digits[i])
            else:
                n2.append(digits[i])
        
        def cal(nums):
            res = 0
            for n in nums:
                res = res * 10 + n
            return res
            
        return cal(n1) + cal(n2)